configuration/configuration_reference.md

```diff
@@ -877,6 +857,211 @@ can be used to **customize how config repo plugin works when parsing this specif
+| port | Yes | Perforce server connection to use (host:port). This is the same as you would pass in the p4port parameter for the p4 command line or in the P4PORT environment variable. |
+| username | No | Perforce username to use. |
+| password | No | Password for the specified user. |
+| encryptedPassword | No | Encrypted Password for the specified user. |
+| useTickets | No | Set to true to work with perforce tickets. Go will do a p4 login using the supplied password before each command. We recommend that you make your user a part of a p4 group, and set the ticket timeout to unlimited as described [here](http://www.perforce.com/perforce/doc.current/manuals/cmdref/login.html). |
+| view | Yes | Valid Perforce view. The view should be a sub-element of P4. Click [here](http://www.perforce.com/perforce/doc.082/manuals/p4guide/02_config.html#1066090) to see details about VIEW of Perforce. |
+| materialName | Required if this material is referenced in pipeline labeltemplate | The name to identify a material. Material name can contain the following characters: a-z, A-Z, 0-9, fullstop, underscore and hyphen. Spaces are not allowed. Material name is case insensitive. It needs to be unique within a pipeline. The max length is 255 characters. |
+| autoUpdate | No | By default Go polls the repository for changes automatically. If autoUpdate is set to false then Go will not poll the repository for changes. Instead it will check for changes only when you trigger a pipeline that contains this material. If the same material is specified more than once in the configuration file, all of them must have the same value for autoUpdate. |
+
+### Notes:
+
+You do not need to specify the above attributes if you have already defined them as system variables. So if you have a P4PASSWD variable
+defined then you can leave out the "password" tag defined above. If you already have them defined as system variables and also in Go
+configuration, Go will overwrite them before running p4.
+
+Views consist of multiple mappings. Each mapping has two parts:
+
+1.  The left-hand side specifies one or more files in the depot and has the form: //depotname/file\_specification
+2.  The right-hand side specifies one or more files in the client workspace and has the form: //clientname/file\_specification
+
+Go creates a p4 client to check out files into its sandbox with the 'clobber' option set. This means, during material update all writable-but-unopened files in the workspace would be overwritten on the agent. All other options use default values as defined by Perforce. Client name is generated automatically by Go. Hence, you can use anything as 'clientname' on the right-hand side in view mapping. The client name format is: cruise-[hostname]-[pipeline name]-[a random hash code], for example "cruise-myhostname-mypipelinename-wOaJ9kjpfgOLQCncki19ikXt5Q". THE GO\_P4\_CLIENT environment variable will have the client name used. This variable can be used in scripts to get the client name
```

Comment:
Remove `THE GO\_P4\_CLIENT environment variable will have the client name used. This variable can be used in scripts to get the client name`, doesn't make sense in this context - there is no client side script